packages/components/src/slot-fill/provider.js

```diff
@@ -94,17 +100,13 @@ class SlotFillProvider extends Component {
 	}
 
 	render() {
-		return this.props.children;
+		return (
+			<Provider value={ this.state }>
```

Comment:
No, they don't. This needs to be an object lifted up to the constructor, to prevent re-render consumers every time this component renders. I don't have any preference on use `state` on a different object.

Reply:
It doesn't matter that much. It needs to be passed here by reference to avoid re-renders as @nosolosw pointed already.